File: exercises/python/checkout/receipt.py
class Receipt:
    def __init__(self):
        self.text = ""
        self.total = 0
        self.a_countdown = 3
        self.b_countdown = 2

    def scanned_a(self):
        self.text += "A: 50"
        self.a_countdown -= 1
        if self.a_countdown == 0:
            self.text += " - 20 (3 for 130)"
            self.total += 30
        else:
            self.total += 50
        self.text += "\n"

    def scanned_b(self):
        self.text += "B: 30"
        self.b_countdown -= 1
        if self.b_countdown == 0:
            self.text += " - 15 (2 for 45)"
            self.total += 15
        else:
            self.total += 30
        self.text += "\n"

    def scanned_c(self):
        self.text += "C: 20\n"
        self.total += 20

    def scanned_d(self):
        self.text += "D: 15\n"
        self.total += 15

    def get_text(self):
        return '{text}Total: {total}'.format(text=self.text, total=self.total)

File: exercises/typescript/src/checkout/receipt.js
class Receipt {
  constructor() {
    this.text = '';
    this.aCountdown = 3;
    this.total = 0;
    this.bCountdown = 2;
  }

  getText() {
    return this.text + "Total: " + this.total;
  };

  scannedA() {
    this.text += 'A: 50';
    if (--this.aCountdown == 0) {
      this.text += ' - 20 (3 for 130)';
      this.total += 30;
    } else {
      this.total += 50;
    }
    this.text += '\n';
  };

  scannedB() {
    this.text += 'B: 30';
    if (--this.bCountdown == 0) {
      this.text += ' - 15 (2 for 45)';
      this.total += 15;
    } else {
      this.total += 30;
    }
    this.text += '\n';
  };

  scannedC() {
    this.text += 'C: 20\n';
    this.total += 20;
  };

  scannedD() {
    this.text += 'D: 15\n';
    this.total += 15;
  };
};

module.exports = {
  Receipt
};